src/comments.js

```diff
@@ -412,19 +416,27 @@ function printComments(path, print, options) {
             parent
           )
         );
+      } else {
+        // Those two node types print their dangling comments themselves
+        if (value.type !== "Program" && value.type !== "BlockStatement") {
+          trailingParts.push(
+            printDanglingComment(
+              commentPath,
+              print,
+              options
+            )
+          );
+        }
```

Comment:
Ooh, that's what the comment is about. I think it's better to go ahead and make the `printDanglingComments` API as simple as possible and go through `printer.js` and add them where appropriate. Most places won't be able to automatically print them; the JSX case is an exception because it still creates an empty node.

What if we made it: `printDanglingComments(path, tabWidth)` ? If you pass in a tab width, it'll indent it that much. We shouldn't need to pass `print` anymore. This API could automatically check if the node has comments, so you literally just need the above code whereever there may be dangling comments.